### PR TITLE
[master] Add script processor

### PR DESCRIPTION
**Pull request by alhambrav**
### Ticket reference or full description of what's in the PR
Add script processor #4443

### CODE REVIEW

File: source/system-administrators/deployer/processors-guide.rst [hunk 1]
```diff
@@ -232,6 +232,35 @@ Processor that updates the processed commits value with the current commit
 
     - processorName: gitUpdateCommitIdProcessor
 
+.. _deployer-script-processor:
+
+^^^^^^^^^^^^^^^^
+Script Processor
```
Comment (sumerjabri) [inline source/system-administrators/deployer/processors-guide.rst:238]: Groovy Script Processor

File: source/system-administrators/deployer/processors-guide.rst [hunk 2]
```diff
@@ -232,6 +232,35 @@ Processor that updates the processed commits value with the current commit
...
+^^^^^^^^^^^^^^^^
+
+Processor that executes scripts in a sandbox
```
Comment (sumerjabri) [inline source/system-administrators/deployer/processors-guide.rst:241]: A site-specific custom Groovy processor that can process published content.

File: source/system-administrators/deployer/processors-guide.rst [hunk 3]
```diff
@@ -232,6 +232,35 @@ Processor that updates the processed commits value with the current commit
...
++------------+-----------+-------------------------------+------------------------------------------------------------+
+
+  .. note::  The default path scripts are loaded from is ``$CRAFTER_HOME/bin/crafter-deployer/processors/scripts``
+
+
+**Example**
+
+.. code-block:: yaml
+    :linenos:
+    :caption: *Script Processor*
```
Comment (sumerjabri) [inline source/system-administrators/deployer/processors-guide.rst:258]: Groovy Script Processor

**Review by sumerjabri — CHANGES_REQUESTED** [review]

**Review by sumerjabri — CHANGES_REQUESTED** [review]
Please reference: https://javadoc.craftercms.org/3.1.11/deployer/org/craftercms/deployer/impl/processors/AbstractMainDeploymentProcessor.html

To show how to implement the Processor.

File: source/system-administrators/deployer/processors-guide.rst [hunk 4]
```diff
@@ -232,6 +232,39 @@ Processor that updates the processed commits value with the current commit
 
     - processorName: gitUpdateCommitIdProcessor
 
+.. _deployer-script-processor:
+
+^^^^^^^^^^^^^^^^^^^^^^^
+Groovy Script Processor
+^^^^^^^^^^^^^^^^^^^^^^^
+
+A site-specific custom Groovy processor that can process published content.
```
Comment (jdrossl) [inline source/system-administrators/deployer/processors-guide.rst:241]: I'm not sure `site-specific` is accurate here, both the processor and the scripts can be shared

File: source/system-administrators/deployer/processors-guide.rst [hunk 5]
```diff
@@ -232,6 +232,39 @@ Processor that updates the processed commits value with the current commit
...
+
+.. code-block:: yaml
+    :linenos:
+    :caption: *Groovy Script Processor*
+
+    - processorName: scriptProcessor
+      scriptPath: 'myscripts/mychanges.groovy'
+
+|
+
```
Comment (jdrossl) [inline source/system-administrators/deployer/processors-guide.rst:264]: For better guidance to developers I think we should include a small example of a valid Groovy script and a short description of the variables available to use

**Review by jdrossl — CHANGES_REQUESTED** [review]

**Review by jdrossl — APPROVED** [review]

**Review by sumerjabri — APPROVED** [review]

**Pull request merged**